src/mptool.f90

```diff
@@ -132,7 +152,134 @@ real(mytype) function  pmax_mytype(var)
+    !
+    ! arguments
+    real(mytype),intent(in) :: var(:)
+    !
+    ! local data
+    integer :: ierr
+    !
+    call mpi_allreduce(var,pmax_mytype_array,size(var),real_type,mpi_max,    &
+                                                    mpi_comm_world,ierr)
+    !
+  end function pmax_mytype_array
+
+  real(mytype) function  pmin_mytype_array(var)
+    !
+    ! arguments
+    real(mytype),intent(in) :: var(:)
+    !
+    ! local data
+    integer :: ierr
+    !
+    call mpi_allreduce(var,pmin_mytype_array,size(var),real_type,mpi_min,    &
+                                                    mpi_comm_world,ierr)
+    !
+  end function pmin_mytype_array
+  real(mytype) function  pmin_mytype(var)
+    !
+    ! arguments
+    real(mytype),intent(in) :: var
+    !
+    ! local data
+    integer :: ierr
+    !
+    call mpi_allreduce(var,pmin_mytype,1,real_type,mpi_min,    &
+                                                    mpi_comm_world,ierr)
+    !
+  end function pmin_mytype
+  !
+  !+-------------------------------------------------------------------+
+  !| this function is to update table based on alltoall mpi            |
+  !+-------------------------------------------------------------------+
+  !| CHANGE RECORD                                                     |
+  !| -------------                                                     |
+  !| 17-Jun-2022  | Created by J. Fang STFC Daresbury Laboratory       |
+  !+-------------------------------------------------------------------+
+  function ptable_update_int_arr(vain) result(vout)
+    !
+    use mpi
+    !
+    integer,intent(in) :: vain(:)
+    integer :: vout(size(vain))
+    !
+    ! local variables
+    integer :: nvar,ierr
+    !
+    nvar=size(vain)
+    !
+    call mpi_alltoall(vain,1,mpi_integer,                   &
+                      vout,1,mpi_integer,mpi_comm_world,ierr)
+    !
+    return
+    !
+  end function ptable_update_int_arr
   !
+  function updatable_int(var,offset,debug,comm,comm_size) result(table)
+    !
+    use mpi
+    !
+    ! arguments
+    integer,allocatable :: table(:)
+    integer,intent(in) :: var
+    integer,optional,intent(out) :: offset
+    logical,intent(in),optional :: debug
+    integer,intent(in),optional :: comm,comm_size
+    !
+    ! local data
+    integer :: comm2use,comm2size
+    integer :: ierr,i
+    integer,allocatable :: vta(:)
+    logical :: ldebug
+    !
+    if(present(debug)) then
+      ldebug=debug
+    else
+      ldebug=.false.
+    endif
+    !
+    if(present(comm)) then
+        comm2use=comm
+    else
+        comm2use=mpi_comm_world
+    endif
+    !
+    if(present(comm_size)) then
+        comm2size=comm_size
+    else
+        comm2size=nproc
+    endif
+    !
+    allocate(table(0:comm2size-1),vta(0:comm2size-1))
+    !
+    call mpi_allgather(var,1,mpi_integer,                              &
+                       vta,1,mpi_integer,comm2use,ierr)
+    !
+    table=vta
```

Comment:
The temporary variable `vta` seems useless. Is ti possible to use directly `table` ?

Reply:
done.